srcs/response/Response.cpp

```diff
@@ -730,44 +727,52 @@ bool	Response::checkRequest()
 			_req_parsed["Version"] != "HTTP/2.0")
 	{
 		std::cerr << "Error 505 Http Version not supported" << std::endl;
-		printError("505", "HTTP Version Not Supported");
+		_response = ErrorResponse::getErrorResponse(ERROR_505, _configs.
+			getErrorPage(ERROR_505));
 		return (1);
 	}
-	if (_configs.getHttpMethods().find(_req_parsed["Method"]) == _configs.
+	if (!_configs.getHttpMethods().empty() && _configs.getHttpMethods().find
+		(_req_parsed["Method"]) == _configs.
 		getHttpMethods().end())
 	{
 		std::cerr << "Error 405 invalid method" << std::endl;
-		printError("405", "Method Not Allowed");
+		_response = ErrorResponse::getErrorResponse(ERROR_405, _configs.
+			getErrorPage(ERROR_405));
 		return (1);
 	}
 	return (0);
 }
 void	Response::init()
 {
-	_error_flag = 0;
 	/*Iniciando o server com os dados do path selecionado*/
-	if (_req_parsed["Path"] == "/")
+	std::string url;
+	std::string _clean_address;
+
+	url = _req_parsed["Path"].substr(0,_req_parsed["Path"].find('/', 1));
+	if (_req_parsed["Path"] != url)
+		_clean_address =  _req_parsed["Path"].substr(_req_parsed["Path"].find
+			('/', 1),_req_parsed["Path"].size());
```

Comment:
Se eu entendi bem, esse _clean_address é o endereço acessado, tirando a parte que pertence a rota. Por exemplo, a rota está configurada como /pasta, mas a pessoa acessou /pasta/teste. Aí nesse caso, o clean address vai ser somente a string /teste. Se for isso, sugiro mudar "_req_parsed["Path"].size()", na linha 754 para "std::string:: npos", ou até mesmo nada. a linha pode ser:
_clean_address =  _req_parsed["Path"].substr(_req_parsed["Path"].find('/', 1));

Isso porque o método substr por padrão, cria uma sub string até o final sem precisar especificar